src/Microsoft.ML.Data/DataView/CacheDataView.cs

```diff
@@ -682,7 +686,7 @@ public bool Wait(long pos)
             {
                 foreach (var w in _waiters)
                     w.Wait(pos);
-                return pos < _parent._rowCount || _parent._rowCount == -1;
+                return !_parent.GetRowCount().HasValue || pos < _parent.GetRowCount().Value;
```

Comment:
I think we should call GetRowCount only once here too. Otherwise there may be an issue between the first and second calls.

Reply:
This fix works as we have adjust the order of these 2 conditions.
if GetRowCount() return null which means the row count has not been set, return true here, otherwise compare the value of RowCount with position. 

So no issue with this implementation and this read natural as we first check whether RowCount has value then compare value if necessary.

---
In reply to: [378696394](https://github.com/dotnet/machinelearning/pull/4829#discussion_r378696394) [](ancestors = 378696394)

Reply:
I'm in favor of single-call here for readability, even though both are correct.

Reply:
I agree. 

---
In reply to: [379060382](https://github.com/dotnet/machinelearning/pull/4829#discussion_r379060382) [](ancestors = 379060382)

Reply:
Ok, I will made the change as suggested

---
In reply to: [379061447](https://github.com/dotnet/machinelearning/pull/4829#discussion_r379061447) [](ancestors = 379061447,379060382)